server/service/integration_live_queries_test.go

```diff
@@ -100,28 +102,53 @@ func (s *liveQueriesTestSuite) TestLiveQueriesRestOneHostOneQuery() {
 		)
 		require.NoError(t, err)
 
-		s.lq.On("QueriesForHost", uint(1)).Return(map[string]string{fmt.Sprint(q1.ID): "select 1 from osquery;"}, nil)
+		s.lq.On("QueriesForHost", uint(1)).Return(map[string]string{fmt.Sprint(q1.ID): query}, nil)
 		s.lq.On("QueryCompletedByHost", mock.Anything, mock.Anything).Return(nil)
-		s.lq.On("RunQuery", mock.Anything, "select 1 from osquery;", []uint{host.ID}).Return(nil)
+		s.lq.On("RunQuery", mock.Anything, query, []uint{host.ID}).Return(nil)
 		s.lq.On("StopQuery", mock.Anything).Return(nil)
 
-		liveQueryRequest := runLiveQueryRequest{
-			QueryIDs: []uint{q1.ID},
-			HostIDs:  []uint{host.ID},
-		}
-		liveQueryResp := runLiveQueryResponse{}
-
 		wg := sync.WaitGroup{}
 		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			s.DoJSON("GET", "/api/latest/fleet/queries/run", liveQueryRequest, http.StatusOK, &liveQueryResp)
-		}()
 
-		// Give the above call a couple of seconds to create the campaign
-		time.Sleep(2 * time.Second)
+		oneLiveQueryResp := runOneLiveQueryResponse{}
+		liveQueryResp := runLiveQueryResponse{}
+		if newEndpoint {
+			liveQueryRequest := runOneLiveQueryRequest{
+				HostIDs: []uint{host.ID},
+			}
+			go func() {
+				defer wg.Done()
+				s.DoJSON("POST", fmt.Sprintf("/api/latest/fleet/queries/%d/run", q1.ID), liveQueryRequest, http.StatusOK, &oneLiveQueryResp)
+			}()
+		} else {
+			liveQueryRequest := runLiveQueryRequest{
+				QueryIDs: []uint{q1.ID},
+				HostIDs:  []uint{host.ID},
+			}
+			go func() {
+				defer wg.Done()
+				s.DoJSON("GET", "/api/latest/fleet/queries/run", liveQueryRequest, http.StatusOK, &liveQueryResp)
+			}()
+		}
+
+		// For loop, waiting for campaign to be created.
+		var cid string
+		cidChannel := make(chan string)
+		go func() {
+			for {
+				campaigns, err := s.ds.DistributedQueryCampaignsForQuery(context.Background(), q1.ID)
```

Comment:
Sleep a few milliseconds in between calls?

Reply:
That seems reasonable, but why exactly? Will not sleeping slow down the test somehow?

Reply:
There is a DB access in this case, so I'm not concerned about CPU load here. But yes, sleep makes sense for CPU-intensive code.